caravel/assets/javascripts/explorev2/actions/exploreActions.js

```diff
@@ -0,0 +1,194 @@
+const $ = window.$ = require('jquery');
+export const SET_DATASOURCE = 'SET_DATASOURCE';
+export const SET_VIZTYPE = 'SET_VIZTYPE';
+export const SET_TIME_COLUMN_OPTS = 'SET_TIME_COLUMN_OPTS';
+export const SET_TIME_GRAIN_OPTS = 'SET_TIME_GRAIN_OPTS';
+export const SET_TIME_COLUMN = 'SET_TIME_COLUMN';
+export const SET_TIME_GRAIN = 'SET_TIME_GRAIN';
+export const SET_SINCE = 'SET_SINCE';
+export const SET_UNTIL = 'SET_UNTIL';
+export const SET_GROUPBY_COLUMNS = 'SET_GROUPBY_COLUMNS';
+export const SET_GROUPBY_COLUMN_OPTS = 'SET_GROUPBY_COLUMN_OPTS';
+export const SET_METRICS = 'SET_METRICS';
+export const SET_METRICS_OPTS = 'SET_METRICS_OPTS';
+export const ADD_COLUMN = 'ADD_COLUMN';
+export const REMOVE_COLUMN = 'REMOVE_COLUMN';
+export const ADD_ORDERING = 'ADD_ORDERING';
+export const REMOVE_ORDERING = 'REMOVE_ORDERING';
+export const SET_TIME_STAMP = 'SET_TIME_STAMP';
+export const SET_ROW_LIMIT = 'SET_ROW_LIMIT';
+export const TOGGLE_SEARCHBOX = 'TOGGLE_SEARCHBOX';
+export const SET_FILTER_COLUMN_OPTS = 'SET_FILTER_COLUMN_OPTS';
+export const SET_WHERE_CLAUSE = 'SET_SET_WHERE_CLAUSE';
+export const SET_HAVING_CLAUSE = 'SET_SET_HAVING_CLAUSE';
+export const ADD_FILTER = 'ADD_FILTER';
+export const SET_FILTER = 'SET_FILTER';
+export const REMOVE_FILTER = 'REMOVE_FILTER';
+export const CHANGE_FILTER_FIELD = 'CHANGE_FILTER_FIELD';
+export const CHANGE_FILTER_OP = 'CHANGE_FILTER_OP';
+export const CHANGE_FILTER_VALUE = 'CHANGE_FILTER_VALUE';
+export const RESET_FORM_DATA = 'RESET_FORM_DATA';
+export const CLEAR_ALL_OPTS = 'CLEAR_ALL_OPTS';
+
+export function setTimeColumnOpts(timeColumnOpts) {
+  return { type: SET_TIME_COLUMN_OPTS, timeColumnOpts };
+}
+
+export function setTimeGrainOpts(timeGrainOpts) {
+  return { type: SET_TIME_GRAIN_OPTS, timeGrainOpts };
+}
+
+export function setGroupByColumnOpts(groupByColumnOpts) {
+  return { type: SET_GROUPBY_COLUMN_OPTS, groupByColumnOpts };
+}
+
+export function setMetricsOpts(metricsOpts) {
+  return { type: SET_METRICS_OPTS, metricsOpts };
+}
+
+export function setFilterColumnOpts(filterColumnOpts) {
+  return { type: SET_FILTER_COLUMN_OPTS, filterColumnOpts };
+}
+
+export function resetFormData() {
+  // Clear all form data when switching datasource
+  return { type: RESET_FORM_DATA };
+}
+
+export function clearAllOpts() {
+  return { type: CLEAR_ALL_OPTS };
+}
+
+export function setFormOpts(datasourceId, datasourceType) {
```

Comment:
i think this function belongs in an `actionCreators` file but it we can change that in a subsequent pr